includes/controllers/class.llms.controller.achievements.php

```diff
@@ -26,48 +36,67 @@ class LLMS_Controller_Achievements {
 	 * @return void
 	 */
 	public function __construct() {
+
+		parent::__construct();
 		add_action( 'init', array( $this, 'maybe_handle_reporting_actions' ) );
 	}
 
 	/**
-	 * Handle achievement form actions to download (for students and admins) and to delete (admins only)
+	 * Returns a translated text of the given type.
 	 *
-	 * @since 3.18.0
-	 * @since 3.35.0 Sanitize `$_POST` data.
+	 * @since [version]
 	 *
-	 * @return void
+	 * @param int   $text_type One of the LLMS_Abstract_Controller_User_Engagements::TEXT_ constants.
+	 * @param array $variables Optional variables that are used in sprintf().
+	 * @return string
 	 */
-	public function maybe_handle_reporting_actions() {
-
-		if ( ! llms_verify_nonce( '_llms_achievement_actions_nonce', 'llms-achievement-actions' ) ) {
-			return;
-		}
+	protected function get_text( $text_type, $variables = array() ) {
 
-		if ( isset( $_POST['llms_delete_achievement'] ) ) {
-			$this->delete( llms_filter_input( INPUT_POST, 'achievement_id', FILTER_SANITIZE_NUMBER_INT ) );
+		switch ( $text_type ) {
+			case self::TEXT_SYNC_AWARDED_ENGAGEMENT_INSUFFICIENT_PERMISSIONS:
+				return sprintf(
+					/* translators: 1: awarded achievement ID */
```

Comment:
This assumes the translator can infer that "%1$d" is the same as "1". Best practice is to use the actual placeholder being used in favor of just the placeholder's number.
```suggestion
					/* translators: %1$d: awarded achievement ID */
```

I am not aware of any specific rule about this but if you look through WP source they seem to exclusively follow this pattern (and that's what we intend to do as well)

Reply:
@thomasplevy,
I agree that the full placeholder is helpful to translators. I was following the examples in [How to Internationalize Your Plugin](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#variables).

There are 423 examples in WordPress code that use just the numbers, but only 2 that use the full placeholder.

Reply:
@pondermatic great, do what I'm asking anyway here <3